tensorly/backend/__init__.py

```diff
@@ -1,21 +1,10 @@
-import sys
-from .. import _BACKEND
+from .core import set_backend, get_backend
+from .core import (context, tensor, is_tensor, shape, ndim, to_numpy, copy,
+                   concatenate, reshape, transpose, moveaxis, arange, ones,
+                   zeros, zeros_like, eye, where, clip, max, min, all, mean,
+                   sum, prod, sign, abs, sqrt, norm, dot, kron, solve, qr, kr,
+                   partial_svd)
 
-sys.stderr.write('Using {} backend.\n'.format(_BACKEND))
-
-if _BACKEND == 'mxnet':
-    from .mxnet_backend import *
-elif _BACKEND == 'numpy':
-    from .numpy_backend import *
-elif _BACKEND == 'pytorch':
-    from .pytorch_backend import *
-elif _BACKEND == 'tensorflow':
-    from .tensorflow_backend import *
-elif _BACKEND == 'cupy':
-    from .cupy_backend import *
-else:
-    import warnings
-    warnings.warn('_BACKEND should be either "mxnet", "pytorch", '
-                  '"tensorflow", "cupy" or "numpy". '
-                  '{} was given.'.format(_BACKEND))
-    warnings.warn('Using MXNet backend.')
+from .core import wrap_module
+wrap_module(__name__)
+del wrap_module
```

Comment:
I am not a big fan of this part: could we potentially explicitly import the default backend here rather than in the core? Or some other alternative that makes explicit what is being done? We could potentially register the backends here? Or do it all in `core.py`?